src/main/webapp/app/overview/course-conversations/dialogs/forward-message-dialog/forward-message-dialog.component.ts

```diff
@@ -0,0 +1,246 @@
+import { UrlAction } from 'app/shared/monaco-editor/model/actions/url.action';
+import { TextEditorAction } from 'app/shared/monaco-editor/model/actions/text-editor-action.model';
+import { MarkdownEditorHeight, MarkdownEditorMonacoComponent } from 'app/shared/markdown-editor/monaco/markdown-editor-monaco.component';
+import { UserPublicInfoDTO } from 'app/core/user/user.model';
+import { CourseManagementService } from 'app/course/manage/course-management.service';
+import { catchError, map, of } from 'rxjs';
+import { ArtemisTranslatePipe } from 'app/shared/pipes/artemis-translate.pipe';
+import { ProfilePictureComponent } from 'app/shared/profile-picture/profile-picture.component';
+import { NgClass } from '@angular/common';
+import { PostingContentComponent } from 'app/shared/metis/posting-content/posting-content.components';
+import { ArtemisSharedModule } from 'app/shared/shared.module';
+import { MetisService } from 'app/shared/metis/metis.service';
+import { LinkPreviewService } from 'app/shared/link-preview/services/link-preview.service';
+import { LinkifyService } from 'app/shared/link-preview/services/linkify.service';
+
+interface CombinedOption {
+    id: number;
+    name: string;
+    type: string;
+    img: string;
+}
+
+@Component({
+    selector: 'jhi-forward-message-dialog',
+    templateUrl: './forward-message-dialog.component.html',
+    styleUrls: ['./forward-message-dialog.component.scss'],
+    imports: [ArtemisTranslatePipe, ProfilePictureComponent, NgClass, PostingContentComponent, MarkdownEditorMonacoComponent, ArtemisSharedModule],
+    providers: [MetisService, LinkPreviewService, LinkifyService],
+})
+export class ForwardMessageDialogComponent implements OnInit, AfterViewInit {
+    channels = signal<ChannelDTO[] | []>([]);
+    users = signal<UserPublicInfoDTO[] | []>([]);
+    postToForward = signal<Post | undefined>(undefined);
+    courseId = signal<number | undefined>(undefined);
+    filteredChannels: ChannelDTO[] = [];
+    filteredUsers: UserPublicInfoDTO[] = [];
+    selectedChannels: ChannelDTO[] = [];
+    selectedUsers: UserPublicInfoDTO[] = [];
+    searchTerm: string = '';
+    isInputFocused: boolean = false;
+    newPost: Post = new Post();
+    defaultActions: TextEditorAction[];
+    editorHeight = input<MarkdownEditorHeight>(MarkdownEditorHeight.INLINE);
+    searchInput = viewChild<ElementRef>('searchInput');
+    messageContent = viewChild<ElementRef>('messageContent');
+    public courseManagementService: CourseManagementService = inject(CourseManagementService);
+
+    showDropdown: boolean = false;
+    combinedOptions: CombinedOption[] = [];
+    filteredOptions: CombinedOption[] = [];
+    showFullForwardedMessage: boolean = false;
+    isContentLong: boolean = false;
+    private cdr = inject(ChangeDetectorRef);
+    public renderer = inject(Renderer2);
+    public activeModal = inject(NgbActiveModal);
```

Comment:
```suggestion
    private activeModal = inject(NgbActiveModal);
```

Reply:
since `activeModal` is used in another file, I converted it to protected